tests/lti/models/auth_data_test.py

```diff
@@ -0,0 +1,121 @@
+            actual = auth_data.get_lti_refresh_token("938200000000000021")
+
+    def test_get_lti_secret(self,auth_data):
+        actual = auth_data.get_lti_secret("93820000000000002")
+        expected = "tJzcNSZadqlHTCW6ow  ... wodX3dfeuIokkLMjrQJqw3Y2"
+        assert actual == expected
+
+    def test_get_lti_secret_throws_key_error(self, auth_data):
+        with pytest.raises(KeyError) as ex:
+            actual = auth_data.get_lti_secret("938200000000000021")
+
+    def test_get_canvas_server_scheme(self,auth_data):
+        actual = auth_data.get_canvas_server_scheme("93820000000000002")
+        expected = "https"
+        assert actual == expected
+
+    def test_get_canvas_server_scheme_throws_key_error(self, auth_data):
+        with pytest.raises(KeyError) as ex:
+            actual = auth_data.get_canvas_server_scheme("938200000000000021")
+
+    def test_get_canvas_server_host(self,auth_data):
+        actual = auth_data.get_canvas_server_host("93820000000000002")
+        expected = "hypothesis.instructure.com"
+        assert actual == expected
+
+    def test_get_canvas_server_host_throws_key_error(self, auth_data):
+        with pytest.raises(KeyError) as ex:
+            actual = auth_data.get_canvas_server_host("938200000000000021")
+
+    def test_get_canvas_server_port(self,auth_data):
+        actual = auth_data.get_canvas_server_port("93820000000000002")
+        expected = "1000"
+        assert actual == expected
+
+    def test_get_canvas_server_port_throws_key_error(self, auth_data):
+        with pytest.raises(KeyError) as ex:
+            actual = auth_data.get_canvas_server_port("938200000000000021")
+
+    def test_get_canvas_server_without_port(self,auth_data):
+        auth_data.auth_data["93820000000000002"]["canvas_server_port"] = None
+
+        actual = auth_data.get_canvas_server("93820000000000002")
+        expected = "https://hypothesis.instructure.com"
+        assert actual == expected
+
+    def test_get_canvas_server_with_port(self,auth_data):
+        actual = auth_data.get_canvas_server("93820000000000002")
+        expected = "https://hypothesis.instructure.com:1000"
+        assert actual == expected
+
+    def test_get_canvas_server_throws_key_error(self, auth_data):
+        with pytest.raises(KeyError) as ex:
+            actual = auth_data.get_canvas_server("938200000000000021")
+
+    def test_set_token(self, auth_data, open_):
+        open_.reset_mock()
+
+        auth_data.set_tokens("93820000000000002", "new_lti_token", "new_refresh_token", open_)
+        assert auth_data.auth_data["93820000000000002"]['lti_token'] == "new_lti_token"
+        assert auth_data.auth_data["93820000000000002"]['lti_refresh_token'] == "new_refresh_token"
+        open_.assert_called_once_with(auth_data.name, 'wb')
+        j = json.dumps(auth_data.auth_data, indent=2, sort_keys=True)
+        open_.return_value.write.assert_called_once_with(j)
+        open_.return_value.close.assert_called_once_with()
```

Comment:
It looks like `set_tokens()` raises `AssertionError` if the `oauth_consumer_key` doesn't exist, could you add a test for that?